### PR TITLE
feat: combine icons and emojis into one menu(#6806)

### PR DESCRIPTION
**Pull request by asjqkkkk**
### Feature Preview

For handling issue #6806

#### Desktop

1. Sidebar Display and Modification
image

2. Article - Table of Contents - Display and Modification
image

3. Article - Title - Display and Modification
image

4. Setting - Move To - Display and Modification
image

5. Linked Page - Display and Modification
image

6. Search Articles - Display
image

#### Mobile

1. Space - Display
image

2. Linked Page -Display
image

3. Emoji Icon - Display
image

4. Recent articles - Display
image

5. Icon Selector
image

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/base/emoji/emoji_picker_screen.dart [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+        child: FlowyIconEmojiPicker(
+          tabs: tabs ?? const [PickerTabType.emoji, PickerTabType.icon],
+          onSelectedEmoji: (r) {
+            context.pop<EmojiIconData>(r);
+          },
+          onSelectedIcon: (r) {
+            context.pop<EmojiIconData>(r.toEmojiIconData());
+          },
+        ),
+      ),
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/base/emoji/emoji_picker_screen.dart:36]:
```suggestion
    return Scaffold(
      appBar: FlowyAppBar(
        titleText: title ?? LocaleKeys.titleBar_pageIcon.tr(),
      ),
      body: SafeArea(
        child: FlowyIconEmojiPicker(
          tabs: tabs,
          onSelectedEmoji: (r) {
            context.pop<EmojiIconData>(r);
          },
          onSelectedIcon: (r) {
            context.pop<EmojiIconData>(r.toEmojiIconData());
          },
        ),
      ),
```

File: frontend/appflowy_flutter/lib/plugins/base/emoji/emoji_picker_screen.dart [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
 import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
 
+import '../../../generated/locale_keys.g.dart';
+import '../../../mobile/presentation/base/app_bar/app_bar.dart';
+import '../../../shared/icon_emoji_picker/tab.dart';
+
 class MobileEmojiPickerScreen extends StatelessWidget {
-  const MobileEmojiPickerScreen({super.key, this.title});
+  const MobileEmojiPickerScreen({super.key, this.title, this.tabs});
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/base/emoji/emoji_picker_screen.dart:11]:
```suggestion
  const MobileEmojiPickerScreen({
    super.key,
    this.title,
    this.tabs = const [PickerTabType.emoji, PickerTabType.icon],
  });
```

File: frontend/appflowy_flutter/lib/plugins/base/emoji/emoji_picker_screen.dart [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
 
   final String? title;
+  final List<PickerTabType>? tabs;
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/base/emoji/emoji_picker_screen.dart:14]:
```suggestion
  final List<PickerTabType> tabs;
```

File: frontend/appflowy_flutter/lib/plugins/database/widgets/row/row_banner.dart [hunk 4]
```diff
@@ -151,7 +151,9 @@ class _RowBannerState extends State<RowBanner> {
                               ? _toolbarHeight - _iconHeight / 2
                               : _toolbarHeight,
                           child: RowIcon(
-                            icon: state.rowMeta.icon,
+                            ///TODO: avoid hardcoding for [FlowyIconType]
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/database/widgets/row/row_banner.dart:154]: Is `FlowyIconType.emoji` correct to use here?

**Comment by LucasXu0** [comment]
Hi, @asjqkkkk. There's a file named `sidebar_icon_test.dart` in the `integration_test` folder. You can add the icon test in this file.

**Comment by annieappflowy** [comment]
image

Test whether the size of the icon is changed per text size by using H1-H3

**Comment by LucasXu0** [comment]
By the way, there are some cases that need to be handled as well.

mention a page
Screenshot 2024-12-17 at 11 44 01

link to page via slash menu
Screenshot 2024-12-17 at 11 45 28

**Comment by asjqkkkk** [comment]
> image
> 
> Test whether the size of the icon is changed per text size by using H1-H3

This case is fine 

image

**Comment by asjqkkkk** [comment]
> By the way, there are some cases that need to be handled as well.
> 
> mention a page Screenshot 2024-12-17 at 11 44 01
> 
> link to page via slash menu Screenshot 2024-12-17 at 11 45 28
> 
> ai chat page Screenshot 2024-12-17 at 11 47 25

I have completed these missing cases, and now I am spending time on writing unit testing.

image

image

**Comment by LucasXu0** [comment]
Need to merge the latest main branch to resovle the CI issue.

**Pull request merged**